### PR TITLE
Update spikekill.php

### PR DESCRIPTION
**Pull request by riversdev0**
This code provides the following minor updates:
1. converts numbers to constants to represent static values
2. minor comment updates
3. additional debug output
4. improved code readability where the html table is generated
5. bug fix: wrong array is sorted when removing low outliers

### CODE REVIEW

File: lib/spikekill.php [hunk 1]
```diff
@@ -286,26 +292,31 @@ private function initialize_spikekill() {
 		}
 
 		switch($this->method) {
-			case 'variance':
```
Comment (riversdev0) [inline lib/spikekill.php:289]: Reorder to match what you see in the Spikekill menu in the GUI

File: lib/spikekill.php [hunk 2]
```diff
@@ -855,12 +872,7 @@ private function calculateOverallStatistics(&$rra, &$samples) {
 							$rra[$rra_num][$ds_num]['avgnksamples']       = 'N/A';
 							$rra[$rra_num][$ds_num]['stddev_killed']      = 'N/A';
 							$rra[$rra_num][$ds_num]['variance_killed']    = 'N/A';
-							$rra[$rra_num][$ds_num]['stddev_killed']      = 'N/A';
 							$rra[$rra_num][$ds_num]['outwind_killed']     = 'N/A';
-							$rra[$rra_num][$ds_num]['numnksamples']       = 'N/A';
-							$rra[$rra_num][$ds_num]['sumnksamples']       = 'N/A';
-							$rra[$rra_num][$ds_num]['variance_killed']    = 'N/A';
-							$rra[$rra_num][$ds_num]['avgnksamples']       = 'N/A';
```
Comment (riversdev0) [inline lib/spikekill.php:863]: Redundant variables with a few lines above.

File: lib/spikekill.php [hunk 3]
```diff
@@ -286,26 +292,31 @@ private function initialize_spikekill() {
...
 				break;
 			case 'fill':
-				$this->method = 4;
+				$this->method = SPIKE_METHOD_FILL;
+				break;
+			case 'float':
+				$this->method = SPIKE_METHOD_FLOAT;
 				break;
 			default:
 				$this->set_error("FATAL: You must specify either 'stddev', 'variance', 'float', or 'fill' as methods.");
```
Comment (riversdev0) [inline lib/spikekill.php:309]: Reorder to match what you see in the Spikekill menu in the GUI

File: lib/spikekill.php [hunk 4]
```diff
@@ -925,16 +937,17 @@ private function outputStatistics($rra) {
...
+								($ds['average'] < 1E6 ? '<td>%s</td>':'<td>%.4e</td>') .
+								($ds['standard_deviation'] < 1E6 ? '<td>%s</td>':'<td>%.4e</td>') .
+								(isset($ds['max_value']) ? ($ds['max_value'] < 1E6 ? '<td>%s</td>':'<td>%.4e</td>') : '<td>%s</td>') .
+								(isset($ds['min_value']) ? ($ds['min_value'] < 1E6 ? '<td>%s</td>':'<td>%.4e</td>') : '<td>%s</td>') .
+								(isset($ds['max_cutoff']) ? ($ds['max_cutoff'] < 1E6 ? '<td>%s</td>':'<td>%.4e</td>') : '<td>%s</td>') .
+								(isset($ds['min_cutoff']) ? ($ds['min_cutoff'] < 1E6 ? '<td>%s</td>':'<td>%.4e</td>') : '<td>%s</td>') .
+								'<td>%s</td><td>%s</td><td>%s</td>' .
+								(isset($ds['avgnksampled']) ? ($ds['avgnksamples'] < 1E6 ? '<td>%s</td>':'<td>%.4e</td>') : '<td>%s</td>') .
+								(isset($ds['variance_avg']) ? ($ds['variance_avg'] < 1E6 ? "<td>%s</td>":"<td>%.4e</td>") : "<td>%s</td>") .
+								"</tr>\n\n",
```
Comment (riversdev0) [inline lib/spikekill.php:950]: Changed the location of the '<td>' tags so that the code is more readable.

File: lib/spikekill.php (hunk 3 above)
Comment (netniV) [reply lib/spikekill.php:309]: Liking the use of constants.

**Review by netniV — CHANGES_REQUESTED** [review]
I thought we were going to target the spikekill changes against 1.3 for proper testing?

**Comment by riversdev0** [comment]
Two things:
1. This pull request doesn't fix any of the open issues with spikekill, it's mainly just a "getting organized" step before actually submitting a pull request for the real issues.
2. Since my understanding of GIT is limited, I've done some things that weren't wise, like requesting to merge my personal '1.2.x' branch into the Cacti `1.2.x' branch, without creating a separate personal branch first, which (i am now seeing) causes different things to get bundled together dynamically (rather than a point-in-time snapshot).  So, my apologies for the 'removespikes.php' changes which are unrelated and which i backed out.

**Review by netniV — APPROVED** [review]
OK, so I did another look over this and it appears to be mainly cursory changes relating to using constants and greater debugging.

**Comment by TheWitness** [comment]
👍

**Pull request merged**